debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/QueueSegment.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.server.ybexporter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.FileDescriptor;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.SyncFailedException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+
+/**
+ * A QueueSegment represents a segment of the cdc queue.
+ */
+public class QueueSegment {
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueueSegment.class);
+
+    private String filePath;
+    private FileOutputStream fos;
+    private FileDescriptor fd;
+    private Writer writer;
+    private long byteCount;
+    private ObjectWriter ow;
+
+    public QueueSegment(String filePath){
+        this.filePath = filePath;
+        ow = new ObjectMapper().writer();
```

Comment:
Let's not use the "new-line delimited json" format. Let's stick to the earlier way of writing pretty-printed json objects. While debugging field issues, we should be able to easily read and understand the event queue files.

Reply:
Discussed offline. For now, going with ndjson. Can revisit if required based on experience. 

(Notes from conversation): 
We thought ndjson is a better choice because:
- It makes the importer processing simpler.
  - With the ndjson
    - to get a json, just read a line and  decode.
    - when to switch to the next queue segment is just about interpreting a specific end-of-file marker like "\.".
  - With the pretty-printed format, this implementation gets a little bit more complicated imo
    - when to switch to the next segment can only be determined by  either detecting the new segment file (or a rename operation like you suggested). Can't detect an EOF marker easily because the json decoder needs to be able to handle cases the marker is a valid json field, etc, etc.
 - As Sanyam pointed out, it adds more whitespaces, and therefore, higher file sizes and potentially some performance loss.